### PR TITLE
tests: restructure e2e_regression

### PR DESCRIPTION
**Pull request by pro-wh**
- run.sh now runs the analysis too. no replacement for compare only, run the whole test instead
- there's no longer a separate "fill cache" target. no replacement, run the whole test instead
- there are no longer separate makefile targets for individual suites. instead run `tests/e2e_regression/run.sh <suite>`
- there are no longer separate makefile targets for accepting individual suites. run `tests/e2e_regression/accept.sh <suite>`
- each suite now has its own test_cases.sh. they source a shared common_test_cases.sh with the overlapping tests
- the symlink-dereferenced "expected" output copy is now scoped by suite name
- CI mistakenly git-diff'd against the non-symlink-dereferenced "expected" output. the code around here is restructured to prevent the two `git diff` invocations from diverging
- run.sh no longer builds nexus. this should be fine, as the test-e2e-regression target depends on the nexus target
- ensure_consistent_config.sh now only runs for one suite at a time. use the new ensure-consistent-config-for-e2e-regression target to run it on all suites

the goal for most changes is to reduce the tasks for starting a new suite as close as possible to copying an existing suite's directory. the additional changes are:
1. add the new suite name to the makefile's `e2e_regression_suites` list
2. add a job for it in ci-test

### CODE REVIEW

File: Makefile
Comment (mitjat) [inline Makefile:91]: One of the reasons we had these as separate targets was that we want to run them as separate (and parallel!) CI jobs.
Example:

Your call on what we do. Ideas:
- we keep it simple and just call sth like `make nexus ensure-consistent-config-for-e2e-regression; ./tests/e2e_regression/run.sh eden`
- we parametrize this make target:

   ```Makefile
   test-e2e-regression: nexus ensure-consistent-config-for-e2e-regression
   ifndef SUITE
   	for suite in $(e2e_regression_suites); do ./tests/e2e_regression/run.sh $$suite; done
   else
   	./tests/e2e_regression/run.sh $(SUITE)
   endif
   ```
   and then call it from CI as `SUITE=eden make test-e2e-regression`.

**Review by mitjat — COMMENTED** [review]
Thank you, looks very nice!

It's currently not running in CI because you're merging into a feature branch; maybe you can rebase onto `main` and temporarily merge this branch into your feature branch? Or maybe you don't need the test there any more anyway because it passes :)

Basically LGTM, just holding off in case you strongly disagree with my comments, or if CI reveals more work is needed.

File: tests/e2e_regression/run.sh
Comment (pro-wh) [reply tests/e2e_regression/run.sh:27]: oo that's a good point, for api work

I'll probably use getopts tho

File: tests/e2e_regression/run.sh
Comment (pro-wh) [reply tests/e2e_regression/run.sh:50]: more options :sunglasses:

File: Makefile
Comment (pro-wh) [reply Makefile:91]: ah yeah ok. or maybe they could set e2e_regression_suites outside

File: tests/e2e_regression/run.sh
Comment (pro-wh) [reply tests/e2e_regression/run.sh:50]: add `-b`

File: tests/e2e_regression/run.sh
Comment (pro-wh) [reply tests/e2e_regression/run.sh]: added `-a`

File: tests/e2e_regression/ensure_consistent_config.sh
Comment (pro-wh) [reply tests/e2e_regression/ensure_consistent_config.sh:24]: no, we have dedicated logic below (`-z "$suite"`) to check if $1 is unset or empty with a custom error message, carried forward from the previous version of run.sh.

also -eu is covered in line 3 already

File: Makefile
Comment (mitjat) [reply Makefile:91]: cool, very elegant :+1:

**Review by mitjat — APPROVED** [review]

File: Makefile
Comment (Andrew7234) [inline Makefile:65]: ah so that's how you do it; i was trying to pass shell env vars to `make` before, with no luck.

**Review by Andrew7234 — APPROVED** [review]
v elegant

File: tests/e2e_regression/run.sh
Comment (pro-wh) [reply tests/e2e_regression/run.sh:50]: eh good motivation but I'm getting the sense from this project that the more normal thing is for make to control scripts, so stuff where scripts control make ought to be the option

File: tests/e2e_regression/run.sh
Comment (pro-wh) [reply tests/e2e_regression/run.sh:9]: changing it to this thx

File: Makefile
Comment (pro-wh) [reply Makefile:65]: huh, but I though I read that env vars work too

**Pull request merged**
